accounts/abi/bind/template.go

```diff
@@ -42,10 +42,28 @@ type tmplContract struct {
 	Fallback    *tmplMethod            // Additional special fallback function
 	Receive     *tmplMethod            // Additional special receive function
 	Events      map[string]*tmplEvent  // Contract events accessors
-	Libraries   map[string]string      // Same as tmplData, but filtered to only keep what the contract needs
+	Libraries   map[string]string      // Same as tmplData, but filtered to only keep direct deps that the contract needs
 	Library     bool                   // Indicator whether the contract is a library
 }
 
+type tmplContractV2 struct {
+	Type        string                 // Type name of the main contract binding
+	InputABI    string                 // JSON ABI used as the input to generate the binding from
+	InputBin    string                 // Optional EVM bytecode used to generate deploy code from
+	Constructor abi.Method             // Contract constructor for deploy parametrization
+	Calls       map[string]*tmplMethod // All contract methods (excluding fallback, receive)
+	Events      map[string]*tmplEvent  // Contract events accessors
+	Libraries   map[string]string      // all direct library dependencies
+	Errors      map[string]*tmplError  // all errors defined
+}
+
+type tmplDataV2 struct {
+	Package   string                     // Name of the package to place the generated file in
+	Contracts map[string]*tmplContractV2 // List of contracts to generate into this file
+	Libraries map[string]string          // Map of the contract's name to link pattern
```

Comment:
This is the only thing that's different which warranted addition of `templDataV2`.